src/erlastic_search.erl

```diff
@@ -411,16 +415,16 @@ bulk_index_docs(IndexTypeIdJsonTuples) ->
     bulk_index_docs(#erls_params{}, IndexTypeIdJsonTuples).
 
 %% Documents is [ {Index, Type, Id, Json}, {Index, Type, Id, HeaderInformation, Json}... ]
+%% prepends 'index' as operation and calls bulk_operation fun
 -spec bulk_index_docs(#erls_params{}, list()) -> {ok, list()} | {error, any()}.
 bulk_index_docs(Params, IndexTypeIdJsonTuples) ->
-     Body = lists:map(fun
-          Build({Index, Type, Id, Doc}) ->
-               Build({Index, Type, Id, [], Doc});
-          Build({Index, Type, Id, HeaderInformation, Doc}) ->
-               Header = bulk_index_docs_header(Index, Type, Id, HeaderInformation),
-               [ Header, <<"\n">>, maybe_encode_doc(Doc), <<"\n">> ]
-     end, IndexTypeIdJsonTuples),
-     erls_resource:post(Params, <<"/_bulk">>, [], [], iolist_to_binary(Body), Params#erls_params.http_client_options).
+    OperationIndexTypeIdJsonTuples = lists:map(fun
+                                                 Build({Index, Type, Id, Doc}) ->
+                                                   Build({Index, Type, Id, [], Doc});
+                                                 Build({Index, Type, Id, HeaderInformation, Doc}) ->
+                                                   {index, Index, Type, Id, HeaderInformation, Doc}
+                                               end, IndexTypeIdJsonTuples),
```

Comment:
A more elegant way to do the same thing would be to have an `operation()` being of the form `{index | create | update | delete, metadata_tuple()}` which would make a lot of sense given how ES thinks of those (see the official doc); and then this fun would simply be
```
bulk_index_docs(Params, IndexTypeIdJsonTuples) ->
   bulk_operation(Params, [{index, IndexTypeIdJsonTuple} || IndexTypeIdJsonTuple <- IndexTypeIdJsonTuples]).